cmd/job_refresh.go

```diff
@@ -154,14 +167,57 @@ func refreshJobSpecificationRequest(l log.Logger, projectName string, namespaces
 		l.Info(coloredSuccess("Refreshed %d jobs.", refreshSuccessCounter))
 	}
 
-	if len(deployErrors) > 0 {
-		l.Error(coloredError("Deployed %d/%d jobs.", deploySuccessCounter, deploySuccessCounter+deployFailedCounter))
-		for _, reqErr := range deployErrors {
-			l.Error(coloredError(reqErr))
+	return "", streamError
+}
+
+func pollJobDeployment(l log.Logger, deployID string, host string) (isFinished bool, err error) {
+	dialTimeoutCtx, dialCancel := context.WithTimeout(context.Background(), OptimusDialTimeout)
+	defer dialCancel()
+
+	var conn *grpc.ClientConn
+	if conn, err = createConnection(dialTimeoutCtx, host); err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			l.Error(ErrServerNotReachable(host).Error())
+		}
+		return false, err
+	}
+	defer conn.Close()
+
+	refreshTimeoutCtx, deployCancel := context.WithTimeout(context.Background(), refreshTimeout)
+	defer deployCancel()
+
+	jobSpecService := pb.NewJobSpecificationServiceClient(conn)
+	resp, err := jobSpecService.GetDeployJobsStatus(refreshTimeoutCtx, &pb.GetDeployJobsStatusRequest{
+		DeployId: deployID,
+	})
+	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			l.Error(coloredError("Get deployment process took too long, timing out"))
+		}
+		return false, fmt.Errorf("getting deployment status failed: %w", err)
+	}
+
+	switch resp.Status {
+	case models.JobDeploymentStatusInProgress.String():
+		l.Info("Deployment is in progress...")
+		return false, nil
+	case models.JobDeploymentStatusInQueue.String():
+		l.Info("Deployer is busy. Deployment request is still in queue...")
+		return false, nil
+	case models.JobDeploymentStatusCancelled.String():
+		l.Error("Deployment request is cancelled. Deployer queue might be full.")
+		return false, nil
```

Comment:
yes, it should be true.

Reply:
updated. however, since we are moving the polling iteration logic inside the `pollJobDeployment` function, the return for isFinished is no longer needed.